stdlib/public/runtime/ProtocolConformance.cpp

```diff
@@ -449,51 +449,90 @@ void swift::_swift_initializeCallbacksToInspectDylib(
 # error No known mechanism to inspect dynamic libraries on this platform.
 #endif
 
-// This variable is used to signal when a cache was generated and
-// it is correct to avoid a new scan.
-static unsigned ConformanceCacheGeneration = 0;
-
 void
 swift::swift_registerProtocolConformances(const ProtocolConformanceRecord *begin,
                                           const ProtocolConformanceRecord *end){
   auto &C = Conformances.get();
   _registerProtocolConformances(C, begin, end);
 }
 
-/// Search the witness table in the ConformanceCache. \returns a pair of the
-/// WitnessTable pointer and a boolean value True if a definitive value is
-/// found. \returns false if the type or its superclasses were not found in
-/// the cache.
+
+struct ConformanceCacheResult {
+  // true if witnessTable is an authoritative result as-is.
+  // false if more searching is required (for example, because a cached
+  // failure was returned in failureEntry but it is out-of-date.
+  bool isAuthoritative;
+
+  // The matching witness table, or null if no cached conformance was found.
+  const WitnessTable *witnessTable;
```

Comment:
This turns out to be annoying because struct WitnessTable is an incomplete type and PointerIntPair wants to call alignof on it.